src/modules/dashboard/components/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferFundsSection.tsx

```diff
@@ -40,47 +62,142 @@ interface Props {
   values: FormValues;
   transactionFormIndex: number;
   setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void;
+  customAmountError: MessageDescriptor | string | undefined;
+  handleCustomAmountError: React.Dispatch<
+    React.SetStateAction<MessageDescriptor | string | undefined>
+  >;
 }
 
 const renderAvatar = (address: Address, item: AnyUser) => (
   <UserAvatar address={address} user={item} size="xs" notSet={false} />
 );
 
 const TransferFundsSection = ({
-  colony: { tokens, colonyAddress },
+  colony: { colonyAddress, safes },
   disabledInput,
   values,
   transactionFormIndex,
   setFieldValue,
+  customAmountError,
+  handleCustomAmountError,
 }: Props) => {
+  const [safeBalances, setSafeBalances] = useState<SafeBalance[] | null>([]);
+  const [isLoadingBalances, setIsLoadingBalances] = useState<boolean>(true);
   const { data: colonyMembers } = useMembersSubscription({
     variables: { colonyAddress },
   });
-  const { feeInverse: networkFeeInverse } = useNetworkContracts();
-  const selectedTokenAddress =
-    values.transactions[transactionFormIndex].tokenAddress;
-  const selectedToken = useMemo(
-    () => tokens.find((token) => token.address === selectedTokenAddress),
-    [tokens, selectedTokenAddress],
+
+  const safeAddress = values.safe?.profile?.walletAddress;
+  const { amount } = values.transactions[transactionFormIndex];
+
+  const getSafeBalance = useCallback(async () => {
+    if (values.safe?.profile?.displayName) {
+      setIsLoadingBalances(true);
+      try {
+        const chainName = getChainNameFromSafe(values.safe.profile.displayName);
+        const baseUrl = getTxServiceBaseUrl(chainName);
+        const response = await fetch(
+          `${baseUrl}api/v1/safes/${safeAddress}/balances/`,
```

Comment:
Note that this will not need 'api' after you rebase as this has been included in the baseUrl.